File: ProjetoBenchMarkCarros/Models/Usuario.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Threading.Tasks;

namespace ProjetoBenchMarkCarros.Models
{
    public class Usuario
    {
        [Key]
        public int IdUsuario { get; set;}
        [Required]
        public string? NomeUsuario { get; set; }
        [Required]
        public string? SenhaUsuario {get; set; }
    }
}

File: ProjetoBenchMarkCarros/front/src/Models/Usuario.ts
// src/models/Usuario.ts

export class Usuario {
  idUsuario: number;
  nomeUsuario: string;
  senhaUsuario: string;

  constructor(data: Partial<Usuario> = {}) {
    this.idUsuario = data.idUsuario ?? 0;
    this.nomeUsuario = data.nomeUsuario ?? '';
    this.senhaUsuario = data.senhaUsuario ?? '';
  }
}
